cluster_pack/uploader.py

```diff
@@ -174,26 +179,33 @@ def _get_hash(spec_file: str) -> str:
 
 def _upload_zip(
     zip_file: str, package_path: str,
-    resolved_fs: Any = None, force_upload: bool = False
+    resolved_fs: filesystem.EnhancedFileSystem, force_upload: bool = False
 ) -> None:
     packer = packaging.detect_packer_from_file(zip_file)
     if packer == packaging.PEX_PACKER and resolved_fs.exists(package_path):
-        with tempfile.TemporaryDirectory() as tempdir:
-            local_copy_path = os.path.join(tempdir, os.path.basename(package_path))
-            resolved_fs.get(package_path, local_copy_path)
-            info_from_storage = PexInfo.from_pex(local_copy_path)
-            into_to_upload = PexInfo.from_pex(zip_file)
-            if not force_upload and info_from_storage.code_hash == into_to_upload.code_hash:
-                _logger.info(f"skip upload of current {zip_file}"
-                             f" as it is already uploaded on {package_path}")
-                return
+        if force_upload:
+            _logger.info(f"forcing upload: removing current {package_path}")
+            resolved_fs.rm(package_path)
+        elif not _check_pex_version(resolved_fs, zip_file, package_path):
+            raise FileExistsError(f"Package {zip_file} already exists"

```

Comment:
If the zip file exists with a different hash we need to override it. Otherwise we could just remove the file content checking and say that the pex file name is always unique when the content changes and in this case we just upload a new file name (removal not handled though)